### PR TITLE
A system to build multiple MooseDocs configurations simultaneously

### LINKED ISSUE #18137: A system to build multiple MooseDocs configurations simultaneously
## Reason
MooseDocs generates HTML pages by reading a configuration file `config.yml` and creating a translator object, which creates reader, extension, renderer, and executioner objects accordingly. However, the vast array of capabilities provided by this system  is currently limited to a single set of configuration options for any given build job. @aeslaughter and I seek a method to read multiple `config.yml` files in a single build job and create a shared pool of content so that every markdown page is translated in the appropriate way, but also so that they may cross-link between each other using standard MooseDown syntax.

## Design
After `load_config.py` reads a configuration file in the normal way and returns a translator object, we can also make it read any number of "subconfigurations" and return that many "subtranslator" objects to `build.py`. This will allow for the creation of separate reader, extension, renderer, and executioner objects to translate any given page in the global content pool in its own special way. We can then communicate the pages between all translators before invoking their individual executioner methods where the hyperlinks will be created.

## Impact
Expanding the documentation in this way will allow multiple sets of content with very different configurations, such as the main website and the workshop, be built at the same time and reference each other. For example, we could create links to the main website in the workshop.

### PR DESCRIPTION
**Pull request by crswong888**
@aeslaughter, please take a look at this when you get the chance. To demonstrate the results you can run something like the following, which builds the test site, workshop, and a new test site all at the same time with cross-linking capabilities:

```
cd ~/projects/moose/python/MooseDocs/test
./moosedocs.py build --serve --subconfigs ../../../tutorials/darcy_thermo_mech/doc/config.yml subconfig/config.yml
```

I still need to set up the live server routines and a comprehensive testing system. Please make any requests for changes that you have at this point.

(closes #18137) (refs #18136)

## Post-merge Update

The command listed above is no longer valid. To demo this new feature, run the following instead:

```
cd ~/projects/moose/python/MooseDocs/test
./moosedocs.py build --serve --config config.yml subsite_config.yml
```

Although, all files that end with 'config.yml' are now the default list of configurations, so the above is equivalent to `./moosedocs.py build --serve`.

### CODE REVIEW

**Review by aeslaughter — CHANGES_REQUESTED** [review]
Made some comments on the main changes, this is a great start. Thanks for the work on this, it will be really great when it is done. It will be another strong feature that will set MooseDocs apart from other similar tools.

File: python/MooseDocs/commands/build.py [hunk 1]
```diff
@@ -136,7 +139,7 @@ def _getPage(self, source):
                 page.base = self._translator.get('destination')
                 if isinstance(page, pages.Source):
                     page.output_extension = self._translator.renderer.EXTENSION
-                self._translator.addContent(page)
+                self._translator.addContent(page) # shoudldn't this be addPage()???
```
[the remark this answers is not in the document]
Comment (crswong888) [reply python/MooseDocs/commands/build.py:141]: I'll double-check this once I get into modifying the live server

File: python/MooseDocs/base/executioners.py [hunk 2]
```diff
@@ -123,11 +122,19 @@ def init(self, destination):
 
         LOG.info('Executing extension initPage() methods complete [%s sec.]', time.time() - t)
 
-    def addPage(self, page):
+    def addPage(self, page, set_uid):
         """Add a Page object to be Translated."""
-        page._Page__unique_id = len(self._page_objects)
+        if set_uid:
+            page._Page__unique_id = len(self._page_objects)
```
[the remark this answers is not in the document]
Comment (crswong888) [reply python/MooseDocs/base/executioners.py:128]: we shall handle this one at a later time

**Comment by milljm** [comment]
@crswong888  I am looking into the conda issues. It has to do with the recent use of `mamba`

**Comment by crswong888** [comment]
@milljm No worries! Do you mind if I do another force-push and restart these tests though?

**Comment by crswong888** [comment]
@aeslaughter, this is ready for your review. The top 4 commits are new.

Basically, there's no such thing as a "primary" translator anymore, each are treated equally and the result should be the same no matter what order they are loaded in. Each configuration should specify unique content, otherwise, `load_configs()` will raise an exception, and every translator shares the same exact pool of content. Pretty much everything new I did is in support of this idea that no translator should be special. Additonally, I addressed your comments in the best way that I could.

**Comment by crswong888** [comment]
The live server is now operational.

Additionally, executioners store translator data in dictionaries now and do not modify the Page.uid attribute. Pages may be added during any part of the build procedure and are no longer blocked, including during a live serve. Although, pages added during a live serve of multiple configurations are ignored since it is not possible to determine which translator should be responsible.

It seems that all executioners are now stable and so I have enabled their tests. Whether or not this is because of changes made here is not clear. I am also not sure how to reproduce the steps described in #15887.

**Review by aeslaughter — APPROVED** [review]

**Comment by milljm** [comment]
@crswong888 sorry, it seems I ignored one of your comments! Wasn't intentional.

I am here because this is all very interesting. I am thinking about the ways we can take advantage of this in Civet...

**Comment by crswong888** [comment]
@milljm No worries. Let me know what your ideas are and I can help.

The possibilities for our web design have grown by a lot with this new system. A single page could have its own unique configuration entirely independent from others. This is by no means limited to just slides. For example, take a look at `python/MooseDocs/test/subsite_config.yml`

**Pull request merged**